### ISSUE TITLE
${__searchFilter} seems to be broken for new/updated template variables

### ISSUE THREAD

**Issue by shanson7** [grafana/grafana#60688]
**What happened**: 

When adding or modifying a template variable with `${__searchFilter}` in it, the typeahead autocomplete stops working (no network requests are made).

**What you expected to happen**:

It would send requests to repopulate the autocomplete as you type.

**How to reproduce it (as minimally and precisely as possible)**:

Create/update any template variable with `${__searchFilter}` and then try to use it.

**Anything else we need to know?**:

When editing the JSON directly, the functionality remains working. It seems that the new format of the template variables breks something.

New save looks like:
```
        "query": {
          "queryType": "Default",
          "refId": "A",
          "target": "tag_values(host, namespace=os, name=cpu.percent.idle.g, host=~(?i)${__searchFilter} )"
        },
```
Previously it looked like.
```
        "query": "tag_values(host, namespace=os, name=cpu.percent.idle.g, host=~(?i)${__searchFilter})",
```

This could be related to the changes in https://github.com/grafana/grafana/pull/52341

**Environment**:
- Grafana version: v9.2.4
- Data source type & version: Graphite
- OS Grafana is installed on: RHEL
- User OS & Browser: Windows/Chrome
- Grafana plugins:
- Others:

**Comment by shanson7** [comment]
Probably related (but technically a different bug), the `"definition"` element of the JSON also doesn't seem to get updated, so when looking at the variable screen it looks like the old value is still there.

**Comment by bohandley** [comment]
Hi @shanson7! Thank you for bringing this to our attention. 

For clarification, this is for the new template variable types, **Value Query** and **Metric Name Query**, correct?

I can confirm that these do not interpolate the ${__searchFilter} and will be working on this.

For **Default Query**, this behavior still appears to be working for me. Can you confirm that Default Query still works or is this broken for you as well?

<img width="441" alt="Screen Shot 2023-01-25 at 7 05 47 PM" src="https://user-images.githubusercontent.com/25674746/214722374-37a73373-bd19-4982-8fbf-fd0182801062.png">

**Comment by shanson7** [comment]
> For Default Query, this behavior still appears to be working for me. Can you confirm that Default Query still works or is this broken for you as well?

Definitely broken for `Default Query` for me (see the JSON in the initial bug report `"queryType": "Default"`). This is breaking existing variables as they are modified.

**Comment by bohandley** [comment]
@shanson7 You are correct that the requests are not being made when typing. This behavior I am still looking into. @grafana/dashboards-squad do you know of any changes to template variables that include ${__searchFilter}? When typing in the variable dropdown, no requests are being made. There is some filtering though.

I have been able to successfully create a template variable in https://play.grafana.org/d/000000053/graphite-sumseries-aliasbynode?orgId=1 with the variable `carbon.agents.graphite-0-a.${__searchFilter}`. When typing in the select dropdown, no requests are made but there does appear to be some filtering happening within the dropdown. 

![Image](https://user-images.githubusercontent.com/25674746/215872061-5171c8e1-c473-4b54-ad6a-e5f366350f60.png)

![Image](https://user-images.githubusercontent.com/25674746/215872214-c1d59523-6da4-4bc8-b589-c0c984e54b10.png)

![Image](https://user-images.githubusercontent.com/25674746/215872313-b1f428ad-20c0-4040-973a-62076b6016df.png)

@shanson7 Are. you saying that when you type in the select dropdown there are no results at all being displayed? This might be due to the behavior of the select and how it is filtering without making the requests. There may be too many metrics for it to handle. This is a guess though. What is being returned when you create the variable? Is there a significant amount of metrics?

**Comment by shanson7** [comment]
> @shanson7 Are you saying that when you type in the select dropdown there are no results at all being displayed? This might be due to the behavior of the select and how it is filtering without making the requests. There may be too many metrics for it to handle. This is a guess though. What is being returned when you create the variable? Is there a significant amount of metrics?

Yes, sorry if I was unclear but this is a large result dataset. There is a 10,000 result limit when using the `tag_values` function. We use the `searchFilter` feature to make further requests to narrow down the dataset. I imagine this is the only use for `searchFilter` as smaller sets work just fine with client side filtering.

**Comment by bohandley** [comment]
Hi @grafana/dashboards-squad! Template variables, which I see as owned by dashboards frontend, no longer support $__searchFilter to send requests to filter as a user types. The select dropdown in template variables uses client side filtering instead of filtering by sending requests to the datasource. This is not helpful for users with large datasets (see above comment). How can we help users with large datasets? Should we open an issue to support $__searchFilter by adding a feature to template variables to support it or just support it? I would guess that any users who include $__searchFilter in their template variable queries would have the dropdown repopulated not by client-side filtering and instead by a request to the datasource to narrow down the options. What do you think?

**Comment by polibb** [comment]
Hello 🙂. This is a bug indeed, searching into the select dropdown should request specific results from the datasource. I'll prioritise it on our side and we'll fix it as soon as we can.

**Comment by shanson7** [comment]
> Hello 🙂. This is a bug indeed, searching into the select dropdown should request specific results from the datasource. I'll prioritise it on our side and we'll fix it as soon as we can.

Any update?

**Comment by polibb** [comment]
Hi @shanson7 - yes, I opened a PR - this issue will be closed when the PR is merged 🙂. The issue will be fixed for any 9.5.x releases.
